crates/kubelet/src/container/state.rs

```diff
@@ -20,26 +23,60 @@ pub async fn run_to_completion<S: ResourceState<Manifest = Container, Status = S
     initial_state: impl State<S>,
     shared: SharedState<S::SharedState>,
     mut container_state: S,
-    pod: SharedState<Pod>,
+    pod: Manifest<Pod>,
     container_name: ContainerKey,
 ) -> anyhow::Result<()> {
-    let (pod_name, api) = {
-        let initial_pod = pod.read().await.clone();
-        let namespace = initial_pod.namespace().to_string();
-        let name = initial_pod.name().to_string();
-        let api: Api<KubePod> = Api::namespaced(client.clone(), &namespace);
-        (name, api)
-    };
+    let initial_pod = pod.latest();
+    let namespace = initial_pod.namespace().to_string();
+    let pod_name = initial_pod.name().to_string();
+    let api: Api<KubePod> = Api::namespaced(client.clone(), &namespace);
 
     let mut state: Box<dyn State<S>> = Box::new(initial_state);
 
+    // Forward pod updates as container updates.
+    let initial_container = match initial_pod.find_container(&container_name) {
+        Some(container) => container,
+        None => anyhow::bail!(
+            "Unable to locate container {} in pod {} manifest.",
+            container_name,
+            pod_name
+        ),
+    };
+
+    let (container_tx, container_rx) = Manifest::new(initial_container);
+    let mut task_pod = pod.clone();
+    let task_container_name = container_name.clone();
+    tokio::spawn(async move {
+        while let Some(latest_pod) = task_pod.next().await {
+            let latest_container = match latest_pod.find_container(&task_container_name) {
+                Some(container) => container,
+                None => {
+                    error!(
+                        "Unable to locate container {} in pod {} manifest.",
+                        &task_container_name,
+                        latest_pod.name()
+                    );
+                    continue;
```

Comment:
Should this trigger some sort of error handling? I don't think it is legal in Kubernetes to patch a pod and add a container, which might have triggered this, what other scenarios are there that might cause this to fail?

Reply:
I don't think it is legal, but I don't think it makes sense to do anything beyond logging an error and then continuing to pass on subsequent updates if they are valid. I suppose we could abort the whole Pod but that seems drastic.

Reply:
Yeah, this was an explicit decision if I remember correctly. Aborting the whole pod felt like overkill for something that could fail on a transient network failure